### PR TITLE
submission/2026-03-25_WSD_CosineDecay_Schedule

### PR DESCRIPTION
**Pull request by ShihChunHao**
## Summary

- Replace linear warmdown LR schedule with **Warmup-Stable-Decay (WSD)** cosine schedule
  - 5% warmup → 75% stable at peak LR → 20% cosine decay
  - Prevents premature LR decay, especially critical under step-limited training budgets
- Built on SOTA base (10L, MLP3x, SmearGate, BigramHash 10240, int5/int6, SWA 0.4, zstd-22)
- No existing submission uses cosine decay or WSD schedule

## Preliminary Results (1 GPU, seed=42)

| Config | val_bpb | artifact_bytes |
|--------|---------|---------------|
| 1 GPU, 600s, ~877 steps | 1.2824 | 15,767,236 |

8xH100 3-seed results pending.

## Run Command

```bash
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — submission/2026-03-25_WSD_CosineDecay_Schedule

**BPB:** 1.2824 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `be77e070e76a`, file `records/track_10min_16mb/2026-03-25_WSD_CosineDecay_Schedule/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.03s, dim=512, layers=10, vocab=1024, code=53346 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.03s, dim=512, layers=10, vocab=1024, code=53346 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
